pkg/pillar/cmd/zedagent/handlecontent.go

```diff
@@ -0,0 +1,128 @@
+	items := ctx.pubContentTreeConfig.GetAll()
+	for idStr := range items {
+		found := false
+		for _, cfgContentTree := range cfgContentTreeList {
+			if cfgContentTree.GetUuid() == idStr {
+				found = true
+				break
+			}
+		}
+		// content tree not found, delete
+		if !found {
+			log.Infof("parseContentInfo: deleting %s\n", idStr)
+			unpublishContentTreeConfig(ctx, idStr)
+		}
+	}
+
+	for _, cfgContentTree := range cfgContentTreeList {
+		contentConfig := new(types.ContentTreeConfig)
+		contentConfig.ContentID, _ = uuid.FromString(cfgContentTree.GetUuid())
+		contentConfig.DatastoreID, _ = uuid.FromString(cfgContentTree.GetDsId())
+		contentConfig.RelativeURL = cfgContentTree.GetURL()
+		contentConfig.Format = cfgContentTree.GetIformat()
+		if cfgContentTree.GetIformat() == zconfig.Format_CONTAINER {
+			contentConfig.IsContainer = true
+		}
+		contentConfig.ContentSha256 = cfgContentTree.GetSha256()
+		contentConfig.MaxDownSize = cfgContentTree.GetMaxSizeBytes()
+		contentConfig.DisplayName = cfgContentTree.GetDisplayName()
+		contentConfig.ImageSignature = cfgContentTree.Siginfo.Signature
+		contentConfig.SignatureKey = cfgContentTree.Siginfo.Signercerturl
+
+		// XXX:FIXME certificates can be many
+		// this list, currently contains the certUrls
+		// should be the sha/uuid of cert filenames
+		// as proper DataStore Entries
+		if cfgContentTree.Siginfo.Intercertsurl != "" {
+			contentConfig.CertificateChain = make([]string, 1)
+			contentConfig.CertificateChain[0] = cfgContentTree.Siginfo.Intercertsurl
+		}
+		publishContentTreeConfig(ctx, *contentConfig)
+	}
+	log.Infof("parsing content info config done\n")
+}
+
+func publishContentTreeConfig(ctx *getconfigContext,
+	config types.ContentTreeConfig) {
+	key := config.Key()
+	log.Debugf("publishContentTreeConfig(%s)\n", key)
+	pub := ctx.pubContentTreeConfig
+	pub.Publish(key, config)
+	log.Debugf("publishContentTreeConfig(%s) done\n", key)
+}
+
+func unpublishContentTreeConfig(ctx *getconfigContext, key string) {
+	log.Debugf("unpublishContentTreeConfig(%s)\n", key)
+	pub := ctx.pubContentTreeConfig
+	config, _ := pub.Get(key)
+	if config == nil {
+		log.Errorf("unpublishContentTreeConfig(%s) not found\n", key)
+		return
+	}
+	pub.Unpublish(key)
+	log.Debugf("unpublishContentTreeConfig(%s) done\n", key)
+}
+
+// content tree event watch to capture transitions
+// and publish to zedCloud
+// Handles both create and modify events
+func handleContentTreeStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	status := statusArg.(types.ContentTreeStatus)
+	ctx := ctxArg.(*zedagentContext)
+	uuidStr := status.Key()
+	PublishContentInfoToZedCloud(ctx, uuidStr, &status, ctx.iteration)
+	ctx.iteration++
+}
+
+func handleContentTreeStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	ctx := ctxArg.(*zedagentContext)
+	uuidStr := key
+	PublishContentInfoToZedCloud(ctx, uuidStr, nil, ctx.iteration)
+	triggerPublishDevInfo(ctx)
```

Comment:
Is this to report that more space is available? Or why do you trigger a dev info?

Reply:
Removed.